### PR TITLE
Add new file mechanism

### PR DESCRIPTION
**Pull request by EmmanuelMess**
#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [x] Done  

- Device:LG K11alpha
- OS:7.1.2

#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 1]
```diff
@@ -83,7 +82,12 @@
 import net.schmizz.sshj.sftp.SFTPClient;
 import net.schmizz.sshj.sftp.SFTPException;
 
-/** Hybrid file for handeling all types of files */
+/**
+ * Hybrid file for handeling all types of files
+ *
+ * <p>This is deprecated, please use AmazeFile instead
```
Comment (VishalNehra) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:88]: Whole class is not deprecated yet. 
Also AmazeFile has only the core parts + smb support. Is that the idea here? What's the idea? You're going to migrate all the filesystems helper methods to it?

**Review by VishalNehra — CHANGES_REQUESTED** [review]

File: file_operations/src/main/java/com/amaze/filemanager/file_operations/filesystem/filetypes/AmazeFile.java [hunk 2]
```diff
@@ -0,0 +1,1586 @@
...
+    p = fs.fromURIPath(p);
+    if (File.separatorChar != '/') {
+      p = p.replace('/', File.separatorChar);
+    }
+    this.path = fs.normalize(p);
+    this.prefixLength = fs.prefixLength(this.path);
+  }
+
+  private void loadFilesystem(String path) {
+    if (SmbAmazeFileSystem.INSTANCE.isPathOfThisFilesystem(path)) {
```
[the remark this answers is not in the document]
Comment (EmmanuelMess) [reply file_operations/src/main/java/com/amaze/filemanager/file_operations/filesystem/filetypes/AmazeFile.java:411]: Yes, for this PR only adds SMB support.

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java (hunk 1 above)
Comment (EmmanuelMess) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:88]: Yes. Now helpers should be moved to the new system.

File: file_operations/src/main/java/com/amaze/filemanager/file_operations/filesystem/filetypes/AmazeFile.java [hunk 3]
```diff
@@ -0,0 +1,1586 @@
+/*
+ * Copyright (C) 2014-2014 Arpit Khurana <arpitkh96@gmail.com>, Vishal Nehra <vishalmeham2@gmail.com>,
```
[the remark this answers is not in the document]
Comment (EmmanuelMess) [reply file_operations/src/main/java/com/amaze/filemanager/file_operations/filesystem/filetypes/AmazeFile.java:2]: I have no idea.

**Pull request closed without merging**